src/components/input/SingleLineInputBase.js

```diff
@@ -81,7 +82,10 @@ class SingleLineInputBase extends PureComponent {
       <Box className={classNames} {...boxProps}>
         <div className={theme['input-wrapper']}>
           {connectedLeft}
-          <div className={theme['input-inner-wrapper']}>
+          <div
+            className={theme['input-inner-wrapper']}
+            style={{ width: width && `${width}px`, flex: width && '0 0 auto' }}
```

Comment:
What happens if width is false? Won't you get something like `style={{ width: false, flex: false }}` then?

Reply:
Nice one.

Reply:
Actually, will it not just be evaluated as something that's not supported and "just work" because browsers are awesome?

![](https://puu.sh/CtuoM/d3cdc5631d.png)

Reply:
Probably, but it still makes me cringe.

Reply:
Totally, not saying we should ship it like this :)

Reply:
But `width` will only be `false`, if you assign it that value, right?

Which seems to come close to setting `width` to `false` in CSS yourself, you wouldn't do both things deliberately.

Also in the prop types it will be defined that a boolean is invalid for `width`, so you'd also get an error in the console if you assign such a value.

Reply:
ah, you're right 👍

Reply:
Still, should width be always be passed to flex? I would keep width for width and use a flex prop